app.py

```diff
@@ -1,35 +1,35 @@
 import aws_cdk as cdk
+import src.utils as utils
 
 from src.network_stack import NetworkStack
 from src.ecs_stack import EcsStack
 from src.service_stack import LoadBalancedServiceStack
 from src.load_balancer_stack import LoadBalancerStack
 from src.service_props import ServiceProps
-import src.utils as utils
-
-cdk_app = cdk.App()
 
 # get the environment
 environment = utils.get_environment()
 stack_name_prefix = f"schematic-{environment}"
-image_version = "0.0.11"
 
-# get VARS from cdk.json
-env_vars = cdk_app.node.try_get_context(environment)
-fully_qualified_domain_name = env_vars["FQDN"]
+cdk_app = cdk.App()
+context_vars = cdk_app.node.try_get_context(environment)
+fully_qualified_domain_name = context_vars["FQDN"]
 subdomain, domain = fully_qualified_domain_name.split(".", 1)
-vpc_cidr = env_vars["VPC_CIDR"]
-certificate_arn = env_vars["CERTIFICATE_ARN"]
+vpc_cidr = context_vars["VPC_CIDR"]
+certificate_arn = context_vars["CERTIFICATE_ARN"]
+env_tags = context_vars["TAGS"]
 
-# get secrets from cdk.json or aws parameter store
-secrets = utils.get_secrets(cdk_app)
+# recursively apply tags to all stack resources
```

Comment:
Does it work to tag all the stack resources before defining them (below)?  In the original template, tagging was the final step.